iroh/src/commands/provide.rs

```diff
@@ -108,6 +123,46 @@ pub async fn run(
         )
     };
 
+    let db2 = db.clone();
```

Comment:
Why is the GC task running in the binary directly, and not within the node? I'd say let's move it to there, otherwise it will not run when starting a node embedded via Rust or FFI.
We can add configuration to the builder to enable GC, with something like this maybe?
```
NodeBuilder::with_gc(self, gc: GcConfig)
enum GcConfig {
    Disabled,
    Interval(Duration)
}
```

Reply:
moved it